### PR TITLE
[SPARK-31612][SQL][DOCS][FOLLOW-UP] SQL Reference clean up

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
clean up
1. uncomment out result tables in SQL examples (to be consistent with other SQL examples)
2. Fix small problems: as -> AS, select -> SELECT, statement -> statements
3. Fix several syntax problems. 

### Why are the changes needed?
Fix errors in sql ref

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manually build and check.

### CODE REVIEW

File: docs/sql-ref-syntax-aux-show-databases.md [hunk 1]
```diff
@@ -39,11 +39,12 @@ SHOW { DATABASES | SCHEMAS } [ LIKE regex_pattern ]
...
-      <li>Only <code>*</code> and <code>|</code> are allowed as wildcard pattern.</li>
-      <li>Excluding <code>*</code> and <code>|</code>, the remaining pattern follows the regular expression semantics.</li>
-      <li>The leading and trailing blanks are trimmed in the input pattern before processing. The pattern match is case-insensitive.</li>
-    </ul>
+     <ul>
+       <li>Except for <code>*</code> and <code>|</code> character, the pattern works like a regular expression.</li>
+       <li><code>*</code> alone matches 0 or more characters and <code>|</code> is used to separate multiple different regular expressions,
+       any of which can match. </li>
+       <li>The leading and trailing blanks are trimmed in the input pattern before processing. The pattern match is case-insensitive.</li>
+     </ul>
```
Comment (huaxingao) [inline docs/sql-ref-syntax-aux-show-databases.md:47]: Change to  the same description of ```regex_pattern``` as the one used in SHOW TABLE EXTENDED, SHOW TABLES, SHOW VIEWS.

File: docs/sql-ref-syntax-aux-show-functions.md [hunk 2]
```diff
@@ -60,11 +60,12 @@ SHOW [ function_kind ] FUNCTIONS ( [ LIKE ] function_name | regex_pattern )
...
-      <li>Only <code>*</code> and <code>|</code> are allowed as wildcard pattern.</li>
-      <li>Excluding <code>*</code> and <code>|</code>, the remaining pattern follows the regular expression semantics.</li>
-      <li>The leading and trailing blanks are trimmed in the input pattern before processing. The pattern match is case-insensitive.</li>
-    </ul>
+     <ul>
+       <li>Except for <code>*</code> and <code>|</code> character, the pattern works like a regular expression.</li>
+       <li><code>*</code> alone matches 0 or more characters and <code>|</code> is used to separate multiple different regular expressions,
+       any of which can match. </li>
+       <li>The leading and trailing blanks are trimmed in the input pattern before processing. The pattern match is case-insensitive.</li>
+     </ul>
```
Comment (huaxingao) [inline docs/sql-ref-syntax-aux-show-functions.md:68]: Change to  the same description of ```regex_pattern``` as the one used in SHOW TABLE EXTENDED, SHOW TABLES, SHOW VIEWS.

File: docs/sql-ref-syntax-qry-select-hints.md [hunk 3]
```diff
@@ -78,7 +84,7 @@ SELECT /*+ SHUFFLE_REPLICATE_NL(t1) */ * FROM t1 INNER JOIN t2 ON t1.key = t2.ke
 -- Spark will issue Warning in the following example
 -- org.apache.spark.sql.catalyst.analysis.HintErrorLogger: Hint (strategy=merge)
 -- is overridden by another hint and will not take effect.
-SELECT /*+ BROADCAST(t1) */ /*+ MERGE(t1, t2) */ * FROM t1 INNER JOIN t2 ON t1.key = t2.key;
+SELECT /*+ BROADCAST(t1), MERGE(t1, t2) */ * FROM t1 INNER JOIN t2 ON t1.key = t2.key;
```
Comment (huaxingao) [inline docs/sql-ref-syntax-qry-select-hints.md:87]: ```SELECT /*+ BROADCAST(t1) */ /*+ MERGE(t1, t2) */``` works too, but want to make the example consistent with the syntax ```/*+ join_hint [ , ... ] */```

**Comment by huaxingao** [comment]
cc @maropu

File: docs/sql-ref-identifier.md [hunk 4]
```diff
@@ -64,6 +64,7 @@ Note: If `spark.sql.ansi.enabled` is set to true, ANSI SQL reserved keywords can
 {% highlight sql %}
 -- This CREATE TABLE fails with ParseException because of the illegal identifier name a.b
 CREATE TABLE test (a.b int);
+-- output
```
Comment (maropu) [inline docs/sql-ref-identifier.md:67]: This comment looks fine, but I think we need the same comment in the other error output.

**Review by maropu — APPROVED** [review]
Looks fine

**Comment by maropu** [comment]
btw, how many clean-up PRs are left?

File: docs/sql-ref-identifier.md (hunk 4 above)
Comment (huaxingao) [reply docs/sql-ref-identifier.md:67]: Maybe have the same format as ansi compliance page?
Screen Shot 2020-05-01 at 2 57 21 PM

**Comment by huaxingao** [comment]
> btw, how many clean-up PRs are left?

Two more:
remove all the html syntax following this PR #28414. 
change HTML tables to markdown tables following this PR #28415 (@dilipbiswal might work on this)

File: docs/sql-ref-identifier.md (hunk 4 above)
Comment (maropu) [reply docs/sql-ref-identifier.md:67]: I think adding `-- output` there looks okay, WDYT?
```
INSERT INTO t VALUES ('1');
-- output
  org.apache.spark.sql.AnalysisException: Cannot write incompatible data to table '`default`.`t`':
  - Cannot safely cast 'v': StringType to IntegerType;
```
Could you check documents in the other DBMS-like systems? As the others suggested, I think its better to follow the other document formats.

**Comment by maropu** [comment]
> change HTML tables to markdown tables following this PR #28415 (@dilipbiswal might work on this)

Do you have time to work on it, @dilipbiswal ?

File: docs/sql-ref-identifier.md (hunk 4 above)
Comment (huaxingao) [reply docs/sql-ref-identifier.md:67]: I googled, but didn't have much luck. Only found this one at https://docs.snowflake.com/en/sql-reference/functions/validate.html

Screen Shot 2020-05-01 at 4 08 33 PM

I personally like to either put ```-- output``` or indent 2 spaces for the error message, or both (the way you suggested)

Comment (maropu) [reply docs/sql-ref-identifier.md:67]: cc: @HyukjinKwon @dongjoon-hyun

**Comment by huaxingao** [comment]
I will close this PR. I am changing every sql ref file to remove the html syntax. I will make these changes there.

**Pull request closed without merging**